test/ao/sparsity/test_data_sparsifier.py

```diff
@@ -25,65 +26,91 @@ class TestBaseDataSparsifier(TestCase):
     def test_constructor(self):
```

Comment:
its probably a good idea to start splitting these tests a bit so they don't continue to grow into a supertest.

I'd recommend creating a helper class to run the test which takes as an input a tuple of tensors/params and then does the checks

then have the main tests be test_constructor_parameters and test_constructor_tensors which just passes in the set of tensors or parameters. Otherwise, if the unit tests start to fail it becomes hard to pinpoint the reason, is it because of the parameter handling? tensor? ...etc

Reply:
That's a good point - let me try doing that.

Reply:
I've revamped the unit-tests - please let me know if it looks good